### ISSUE TITLE
Error Messaging on "type of kind OBJECT can not be used as input"

### ISSUE THREAD

**Issue by B-Reif** [graph-gophers/graphql-go#224]
I encountered this error while defining my schema, but I didn't have any idea how to fix it. Eventually I found the error message in `graphql-go`'s source and found this was related to my type being non-nullable.

At least a few other people in #122 had the same issue. Some extra messaging about this case might help.

**Comment by enjoylife** [comment]
I agree that the wording of the error is confusing when first encountering it. 

Is there a sense of how easy it is to update the messaging to encompass the following?

* naming the object in question
* stating if/why it needs  to be nullable
* and steps to fix it

**Comment by pavelnikolov** [comment]
A simple reproducible example would e appreciated.

**Comment by bsr203** [comment]
For future reference. I too spend some time to find the cause, and was the case as in [this](https://github.com/graph-gophers/graphql-go/issues/122#issuecomment-344405512) comment.
